### DISCUSSION TITLE
SMTP Docker Container Exiting Status 1

### DISCUSSION THREAD

**Discussion by mattbaylor** [postalserver/postal#1823] [Help with using Postal] [answered]
I have postal running with the docker configuration, however the container that handles incoming smtp connections keeps exiting with condition 1. Here are the details from `postal logs smtp`

```
smtp_1      | rake aborted!
smtp_1      | 
smtp_1      | Errno::EAFNOSUPPORT: Address family not supported by protocol - socket(2) for "::" port 25
smtp_1      | 
smtp_1      | /opt/postal/app/lib/postal/smtp_server/server.rb:52:in `initialize'
smtp_1      | /opt/postal/app/lib/postal/smtp_server/server.rb:52:in `open'
smtp_1      | /opt/postal/app/lib/postal/smtp_server/server.rb:52:in `listen'
smtp_1      | /opt/postal/app/lib/postal/smtp_server/server.rb:259:in `run'
smtp_1      | /opt/postal/app/lib/tasks/postal.rake:13:in `block (2 levels) in <top (required)>'
smtp_1      | /opt/postal/app/bin/bundle:3:in `load'
smtp_1      | /opt/postal/app/bin/bundle:3:in `<main>'
smtp_1      | 
smtp_1      | Tasks: TOP => postal:smtp_server
smtp_1      | 
smtp_1      | (See full trace by running task with --trace)
```

This feels like an ipv6 issue because of the "::" address family message, but I'm not sure how to resolve. I'm running the docker containers on Ubuntu 20.04. Any ideas?

**Comment by willpower232** [comment, marked as answer]
This happens if IPv6 has been disabled on your server or Postal cannot bind to it. You can change it to only bind on IPv4, have a look at

#468 #847 #1438

**Reply by mattbaylor** [reply]
So it seems I would need to make a change in the postal_smtp docker image if I'm understanding all the linkages correctly?

**Reply by willpower232** [reply]
One of those issues discusses changing the config file so it attaches to `0.0.0.0` and not `::`, that should be the only change you need to help.

**Reply by mattbaylor** [reply]
OK, I was way overcomplicating this. It really is just as simple as adding `bind_address: '0.0.0.0'` to the smtp_server section in `opt/postal/config/postal.yml`. For some reason I had convinced myself I needed to dig into the docker image and change some setting there. Nope. You all have done a fantastic job with this docker based deployment. Thanks again.
